### PR TITLE
Align producer/consumer tensor names so yaml matches ONNX

### PR DESCRIPTION
**Pull request by rafaelcathomen**
Follow-up to #4 and #5. Addresses the recurring deploy-time pain where the generated yaml's tensor names don't match the actual ONNX file's input/output names, forcing the deploy team to hand-edit the yaml after every export.

## Problem

leapp emits the yaml metadata + per-node ONNX files from the same traced graph, but the names disagree.

Concrete example before this PR:

\`\`\`
preprocess_video.onnx actually outputs: ['pixel_values_0', 'input_ids', 'attention_mask', 'image_sizes', 'embodiment_id']
yaml claims preprocess_video outputs:   ['vl_input_pixel_values_0', 'vl_input_input_ids', 'vl_input_attention_mask',
                                         'action_inputs_image_sizes', 'action_inputs_embodiment_id']
\`\`\`

Why: backbone (consumer) takes a dict parameter named \`vl_input\`, so leapp names backbone's inputs \`vl_input_*\`. preprocess_video (producer) outputs the same tensors with their plain dict-key names. leapp's cross-node reconciler rewrites the producer's TensorDescription names to match the consumer — but only in the yaml metadata, **after** the ONNX files have been compiled with the original names.

Same pattern for backbone outputs (\`converted_outputs_*\` in ONNX, \`backbone_outputs_*\` in yaml), preprocess_state (\`state\`/\`reference\` in ONNX, \`action_inputs_state\`/\`state_0_*\` in yaml), and action_head's output (\`output1_action_pred\` in ONNX, \`normalized_action\` in yaml).

## Fix

Pre-shape the producer's output dicts so the names already match the consumer's parameter structure. When alignment holds, leapp's reconciler is a no-op and yaml + ONNX agree by construction.

Specifically:

| Edge | What changed |
|------|--------------|
| preprocess_video → backbone / action_head | Output dict nested as \`{'vl_input': {'pixel_values': ...}}\` and \`{'action_inputs': {...}}\` so leapp produces \`vl_input_*\` / \`action_inputs_*\` names directly |
| preprocess_state → action_head | State tensor wrapped in \`{'action_inputs': {'state': ...}}\` |
| preprocess_state → decode_action | \`reference\` renamed to \`state\` (the param name decode_action uses) |
| backbone return → action_head | Local variable renamed \`converted_outputs\` → \`backbone_outputs\` so leapp uses the consumer's dict-name as the prefix |
| action_head return → decode_action | Returns \`{'normalized_action': ..., ...}\` instead of \`{'action_pred': ...}\` |

## Validation

Tested against a G1 \`NEW_EMBODIMENT\` finetune (332 episodes, full N1.6 export with effort + navigate + base_height outputs).

**Before**: 0 of 5 model files had yaml/ONNX name agreement. Deploy team hand-edited every export.

**After**: 4 of 5 model files have full yaml/ONNX name agreement (preprocess_video, preprocess_state, backbone, decode_action).

Verified with this script:

\`\`\`python
import yaml, onnx
y = yaml.safe_load(open('exported_g1.../<name>.yaml'))
for name, model_def in y['models'].items():
    m = onnx.load(f'.../<name>.onnx', load_external_data=False)
    yaml_io  = sorted([t['name'] for t in model_def['outputs']])
    onnx_io  = sorted([o.name for o in m.graph.output])
    assert yaml_io == onnx_io, f'{name}: yaml says {yaml_io}, onnx says {onnx_io}'
\`\`\`

## Remaining mismatches (action_head)

Two issues that require either a leapp-level fix or a more invasive wrapper refactor:

1. **3 input tensors** (\`pixel_values_0\`, \`input_ids\`, \`attention_mask\`) come from preprocess_video, are consumed by both backbone (via \`vl_input.*\`) and action_head (via \`action_inputs.*\`). When one tensor flows to two consumers under different names, leapp's reconciler picks one for the yaml; the other ONNX file keeps its own name. Fixable by aligning action_head's parameter to match backbone (move these 3 from \`action_inputs\` to \`vl_input\`), but that's a wrapper-signature refactor beyond this PR's scope.

2. **action_head output**: ONNX exports as \`output1_normalized_action\` (leapp prefixes positional return values with \`output1_\`); yaml claims \`normalized_action\` (after reconciler). This is a leapp-internal convention; not fixable from this repo without leapp changes.

Both are acceptable as known limitations for now — deploy team's hand-edit list shrinks from 5 model files to 1.

### CODE REVIEW

**Comment by rafaelcathomen** [comment]
Closing as superseded.

Turns out the right place to fix this is in the **leapp library itself**, not here. The yaml/ONNX name mismatch was caused by leapp running cross-node IO-name reconciliation (`LeappGraph._reconcile_io_names`) *after* `compile_models()` had already exported the ONNX files. The reconciler renamed the in-memory TensorDescriptions to align producer outputs with consumer inputs, but the .onnx files were already written with the pre-reconcile names.

A leapp MR (https://gitlab-master.nvidia.com/Isaac/leapp/-/merge_requests/new?merge_request%5Bsource_branch%5D=rcathomen%2Fsync-onnx-names-with-yaml) reorders the protocol so:

1. `prepare_for_compile_models()` does backend-specific name munging (e.g. ONNX dedup of overlapping IO names from feedback connections)
2. `LeappGraph(...)` reconciles cross-node names
3. `compile_models()` finally calls torch.onnx.export with the now-final names

After the leapp fix, the exporter here doesn't need to know consumer parameter names at all — producers just declare their own outputs. End-to-end test on a G1 finetune confirms 5/5 model files have yaml/ONNX name agreement with leapp fix alone, no frlai changes needed.

Closing this PR; the renames and nested dicts here introduced unnecessary producer→consumer coupling and are no longer needed once leapp ships the fix above. PRs #4 and #5 stay relevant — they fix orthogonal issues (`kind:` values + `register_embodiment_joints` API).
